crypto/bio/b_sock2.c

```diff
@@ -206,7 +208,12 @@ int BIO_listen(int sock, const BIO_ADDR *addr, int options)
     }
 
 # ifdef IPV6_V6ONLY
-    if ((options & BIO_SOCK_V6_ONLY) && BIO_ADDR_family(addr) == AF_INET6) {
```

Comment:
On a *side* note one can wonder if it's appropriate to ignore BIO_SOCK_V6_ONLY in cases when IPV6_V6ONLY is not defined. I mean following. If you have platform that supports AF_INET6, but not IPV6_V6ONLY, then wouldn't it be appropriate to return error to application that attempts BIO_SOCK_V6_ONLY on such platform? Instead of silently ignoring it as it is now that is.

Reply:
I see it as a kind of best effort.

Reply:
> I see it as a kind of best effort.

And question is this is adequate view. I mean if you pass unrecognized option, wouldn't you like to know that it's not supported? Well, one can always argue that AF_INET6 platforms that don't have IPV6_V6ONLY don't deserve respect... For reference, so far I've spotted only one, Windows 2003/XP. So no biggie... Once again, this is a *side* note.